src/schema/type/Annotation.graphql

```diff
@@ -0,0 +1,43 @@
+"http://www.w3.org/ns/oa#TextualBody"
+type TextualBody {
+  "http://www.w3.org/1999/02/22-rdf-syntax-ns#value"
+  value: String!
+  "http://purl.org/dc/elements/1.1/format"
+  format: String
+  "http://purl.org/dc/elements/1.1/language"
+  language: AvailableLanguage
+  # Should always be "http://www.w3.org/ns/oa#TextualBody"
+  "http://www.w3.org/1999/02/22-rdf-syntax-ns#type"
+  type: String
+  "http://purl.org/dc/elements/1.1/identifier,https://schema.org/identifier"
+  identifier: ID
+}
+
+"http://www.w3.org/ns/oa#Annotation"
+type Annotation {
+  # TODO: This should be a UUID or a URL, is there a type that can be used?
```

Comment:
- I think this should be a relationship in the CE, which we can translate to a URL in the JSON-LD output. Otherwise, we can't search by target or show metadata from the target object. If it is an external resource, shouldn't it be added in the CE anyway?
- Are multiple targets allowed? e.g. `target: [URL]` VS `target: URL`
  - https://www.w3.org/TR/annotation-vocab/#hastarget

Reply:
Ideally it should support both (internal) CE identifiers and (external) URIs. If this cannot be implemented due to graphql or neo4j limitations, we could indeed build a service that generates thin wrappers (probably DigitalDocuments) corresponding to external resources. I suspect this could become part of the client-side annotation code...

Multiple targets must be supported. One option to simplify things is to always return a list of targets, even where there is only 1 member. 

Empty lists (0 targets) are not allowed.

Reply:
I think that Christiaan makes a good point here - we don't have to follow the ontology definition in the graphql interface as tightly as I have here. We could have something like an interface to define an AnnotationTarget, which is a union that can be either a link to a thing in the CE, or an external URL (e.g. like a join table in sql). The jsonld generator can then take this and format it correctly. We can also use a similar technique in `motivatedBy` (a standard web annotations motivation, or a custom one that we create) and body (a link to a textualbody or a URL to an external body).
My original idea was to make as few changes as possible to the jsonld converter, but it seems likely that we'll need to have specific converter modules for specific data types. I'll re-think this architecture.